src/main/java/testapp/adapter/http/UserAccountService.java

```diff
@@ -0,0 +1,62 @@
+package testapp.adapter.http;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.google.sitebricks.At;
+import com.google.sitebricks.client.transport.Json;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.http.Get;
+import com.google.sitebricks.http.Post;
+import testapp.core.*;
+
+@At("/accounts")
+public class UserAccountService {
+    private AccountRepository accRepository;
+
+    @Inject
+    public UserAccountService(AccountRepository accRepository) {
+        this.accRepository = accRepository;
+    }
+
+    @Get
+    @At("/:userId/balance")
+    public Reply<?> getUserBalance(@Named("userId") int userId) {
+        int balance;
+        UserAccount account;
+        try {
+            account = accRepository.findUserAccount(userId);
+            balance = account.getBalance();
+        } catch (AccountException | NullPointerException e) {
+            return Reply.saying().badRequest();
+        }
+        return Reply.with(balance).as(Json.class);
+    }
+
+    @Post
+    @At("/:userId/deposit/:value")
```

Comment:
I would suggest that you transmit the value in the body of the request, so that it is not visible in the path of the request